keps/sig-node/2371-cri-pod-container-stats/README.md

```diff
@@ -0,0 +1,416 @@
+Thus, this KEP largely the plan described [here][#plan], with some changes:
+
+- the `/metrics/cadvisor` endpoint will likely not be exposed
+	- **TODO** will this break users?
+- CRI is used for all of the monitoring endpoints related to Containers and Pods (except Volume and Ephemeral Storage)
+- CRI is used to provide metrics for eviction.
+	- **TODO**: This is possible with just the stats Summary API, right?
+
+
+[core-proposal]: https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/core-metrics-pipeline.md#proposed-core-metrics
+[keep-summary]: https://github.com/kubernetes/kubernetes/issues/68522#issuecomment-666636130
+[plan]: https://github.com/kubernetes/kubernetes/issues/68522#issuecomment-724928827
+### Risks and Mitigations
+
+- To properly move to CRI stats, it is likely there are some metrics that we'll want to not support (Accelerator/UserDefined). We should be careful to not break entities that rely on these metrics.
+- A large part of this work is changing the source of the Summary API metrics. In doing so, there is a risk that collecting from a new source will change how the stats look in aggregate (and risk bugs popping up in new areas).
+- cAdvisor has a long history of collecting these stats. There is a risk that changing the source of the stats to the CRI implementation can cause performance regressions
+as cAdvisor is fine tuned to perform in an adequate manner.
+    - CRI implementations should do performance regression analyses to ensure the change does not regress too much.
+
+## Graduation Criteria
+
+
+## Implementation History
+
+
+## Drawbacks [optional]
+
+CRI runtimes will each have to implement additional interface to support full stats, rather than all metric collection being unified by cAdvisor. 
+
+Note: This is by design as this will enable to decouple runtime implementation details further from kubelet.
+
+## Alternatives [optional]
+
+Instead of teaching CRI how to do *everything* cAdvisor does, we could instead have cAdvisor not do the work the CRI stats end up doing (specifically when reporting disk stats, which are the most expensive operation to report). However, this doesn't address the anti-pattern of having multiple parties confusingly responsible for a wide array of metrics and other issues described.
+
+### Notes [optional]
+
+**TODOs**:
+
+* The fields in PodSandboxStats will largely mirror ContainerStats, any room to simplify that?
+* Should we support [User Defined metrics](https://github.com/google/cadvisor/blob/master/docs/application_metrics.md)? This feature is alpha and does not appear to be widely used?
+* There are some fields that are defined in SummaryAPI (e.g time in a lot). Do we add those?
+* How do we tell cAdvisor to not collect container metrics? Below are some options:
+* - Give it a bogus container storage/cgroup root
+* - Flag to cAdvisor
+* - Configuration field in cAdvisor
+* Who collects container log metrics? Kubelet/cAdvisor/CRI?
```

Comment:
cAdvisor does, although the "ephemeral storage" cri container metric should include logs as well.

Reply:
in the proposal, Kubelet is still responsible for Ephemeral storage of the container. We'd have to add the log field to either Kubelet or CRI. I think Kubelet makes more sense because it is the owner of the log directory